src/main/java/pl/debuguj/parkingspacessystem/dao/ParkingSpaceDaoMockImpl.java

```diff
@@ -0,0 +1,117 @@
+import pl.debuguj.parkingspacessystem.domain.ParkingSpace;
+import pl.debuguj.parkingspacessystem.exceptions.CarRegisteredInSystemException;
+import pl.debuguj.parkingspacessystem.exceptions.IncorrectEndDateException;
+import pl.debuguj.parkingspacessystem.exceptions.ParkingSpaceNotFoundException;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Created by grzesiek on 07.10.17.
+ */
+@Service
+public class ParkingSpaceDaoMockImpl implements ParkingSpaceDao {
+
+    private static List<ParkingSpace> listParkingSpaces = new ArrayList<>();
+
+    @Override
+    public void create(final ParkingSpace parkingSpace) throws CarRegisteredInSystemException {
+
+        Optional<ParkingSpace> ps =
+                findByRegistrationNo(parkingSpace.getCarRegistrationNumber());
+        if(!ps.isPresent())
+            listParkingSpaces.add(parkingSpace);
+        else
+            throw new CarRegisteredInSystemException("Car is registered into system");
+    }
+
+    @Override
+    public Optional<ParkingSpace> findByRegistrationNo(final String registrationNo)  {
+        return listParkingSpaces
+                .stream()
+                .filter(ps -> Objects.equals(registrationNo, ps.getCarRegistrationNumber()))
+                .findFirst();
+    }
+
+    @Override
+    public List<ParkingSpace> getAll() {
+        return Collections.unmodifiableList(listParkingSpaces);
+    }
+
+    @Override
+    public Optional<ParkingSpace> updateEndTime(
+            final String registrationNumber,
+            final Date timestamp) throws IncorrectEndDateException, ParkingSpaceNotFoundException {
+
+        Optional<ParkingSpace> ps = findByRegistrationNo(registrationNumber);
+
+        if(ps.isPresent())
+        {
+            if (!timestamp.after(ps.get().getBeginTime())) {
+                throw new IncorrectEndDateException("Incorrect update time");
+            }
+
+            listParkingSpaces
+                    .forEach(s -> {
+                        if (Objects.equals(registrationNumber, s.getCarRegistrationNumber())) {
+                            s.setEndTime(timestamp);
+                        }
+                    });
+
+            return findByRegistrationNo(registrationNumber);
+        }
+        throw new ParkingSpaceNotFoundException("Parking space not found");
+    }
+
+    @Override
+    public void removeAll() {
+        listParkingSpaces.clear();
+    }
+
+    @Override
+    public List<ParkingSpace> findByDate(final Date timestamp) {
+        final Date end = createEndDate(timestamp);
+        return Collections.unmodifiableList(
+                listParkingSpaces
+                .stream()
+                .filter(ps -> timestamp.before(ps.getBeginTime())
+                        && end.after(ps.getBeginTime()))
+                .collect(Collectors.toList()));
+
+    }
+
+    private Date createEndDate(final Date d){
+        return (new DateTime(d).plusDays(1)).toDate();
```

Comment:
co daje ten nawias?

Reply:
Myślałem, że będę używał funkcji kilka razy, a tak naprawdę robię to tylko raz.